crmd/lrm.c

```diff
@@ -1806,9 +1855,36 @@ construct_op(lrm_state_t * lrm_state, xmlNode * rsc_op, const char *rsc_id, cons
     op->timeout = crm_parse_int(op_timeout, "0");
     op->start_delay = crm_parse_int(op_delay, "0");
 
+    rsc = lrm_state_get_rsc_info(lrm_state, rsc_id, 0);
+
+    if (safe_str_eq(op->op_type, CRMD_ACTION_STATUS) && op->interval == 0 && rsc) {
+        lrm_state_update_rsc_metadata(lrm_state, rsc, NULL);
```

Comment:
> This approach actually works because crmd/te_actions.c:te_rsc_command() either calls crmd:/lrmd.c:do_lrm_invoke() or sends the command to the other node.
> The hard part is to inject metadata call in such a way as to ensure that tengine will wait for query to finish before initiating a start action.
> The natural way is to create this action in pengine. But we need to fix a whole bunch of tests then. We need to do this if we change the tengine only, too. But in that case only summary is affected, so we can suppress metadata actions logging in fake_transition.
> What do you think is the preferable way?

I don't have a strong preference. On the one hand, it makes sense that the PE should schedule any action; on the other hand, the metadata call depends on the crmd's cache contents, which is outside the PE's scope. I suppose the PE could always schedule a metadata *check* when metadata is needed, and the crmd could execute a metadata call only if it doesn't already have it cached. As much as I hate the thought of touching nearly every PE test, it might be the most appropriate way.